shared/src/helpers/permissions/check-permission-keys/check-permission-keys.helper.ts

```diff
@@ -0,0 +1,18 @@
+import { PermissionKey } from '~/common/enums/enums';
+
+type Args = {
+  pagePermissions: PermissionKey[];
+  userPermissions: PermissionKey[];
```

Comment:
```suggestion
  permissionKeys: PermissionKey[];
  userPermissions: Permission[];
```
to avoid additional maps